src/string_upto_iterator.cpp

```diff
@@ -0,0 +1,22 @@
+#include "natalie/string_upto_iterator.hpp"
+#include "natalie/integer.hpp"
+
+namespace Natalie {
+const TM::String StringUptoIterator::next() {
+    return m_string = peek();
+}
+
+const TM::String StringUptoIterator::peek() const {
+    if (m_treat_like_integer)
+        return TM::String((Integer(m_string) + 1).to_nat_int_t());
+
+    if (m_string == "9")
+        return ":";
+    else if (m_string == "Z")
+        return "[";
+    else if (m_string == "z")
+        return "{";
```

Comment:
This same `upto` logic is in [src/string.rb](https://github.com/natalie-lang/natalie/blob/master/src/string.rb) ... how could we consolidate?

Reply:
Nevermind! I see you mentioned that in the PR description. Let's merge this and plan to use it from StringObject somehow, later...